beginning of API test refactor and cleanup

cgolubi1 opened the pull request:
Partially addresses #1858 but doesn't fix it entirely yet

* Generalize the framework for loading dummy data from files to support more API methods, and both successes and failures
* Migrate additional API methods to new test framework: createUser, createGame, countPendingGames, adjustFire, createForumThread, createForumPost
* Add a UI loadGameData failure test

Jenkins: http://jenkins.buttonweavers.com:8080/job/buttonmen-cgolubi1/418/ (if it passes)

cgolubi1 commented:
Shadowshade: i'm still actively working on this, but i'm putting it up right now because i don't know what your pull approval availability is, and i want to make sure the approach isn't totally off-base.  I will keep my branch clean except when explicitly noted, so if you like it, merge it (assuming it passes jenkins).

In terms of "when will i be done?", for purposes of #1858 i'll be done when:
* `untransformedFunctions` is empty and can be snipped from the code (i.e. all DummyApiResponder activity takes place using JSON files created by responderTest)
* i can remove the `$this->dummy` object from responderTest (i.e. all existing responderTests are achieving the approximate quality of testing they have now, but are only producing dummy data, not consuming it)

Testing every function is not a goal of this particular project --- the goal is to refactor all API tests which exist now without loss of test quality.

blackshadowshade commented:
I'd like to know what your motivation is for the changes in the unit test values.

cgolubi1 commented:
> I'd like to know what your motivation is for the changes in the unit test values.

Where i did things like change 5 to 1 or some other button to Avis, it's because the UI/python tests and the backend tests weren't linked to each other, so were expecting different values.  So i changed one to match the other.

Where i modified the die values in the fire test, it's because the UI tests wanted a responderTest in which responder003 (typically the active player for data storage purposes) was the active player, had the option to adjust fire dice, and did so.  That didn't exist, so i adjusted some die values in the existing test to make it possible for responder003 to make a fire-assisted attack on its next move, and then added that next move.

blackshadowshade commented:
Okay, this runs fine on my MAMP stack, and produces new test files in the dummy_data directory under new directories as expected. Also, Jenkins passes fine. I'm happy to accept this pull request.
